analytics/upress_reporting/models/reports/downloads.py

```diff
@@ -34,16 +57,63 @@ def build_header(self):
             "Created_By": self.created_by,
         }
 
-    def build_report(self):
-        download_events = self.download_request_parser.pull_download_events()
-        header = self.build_header()
-        with open('counter_5_downloads_report.csv', 'w') as csv_file:
-            writer = csv.writer(csv_file)
-            for key, value in header.items():
-                writer.writerow([key, value])
-            writer.writerow([])
-            # following code is temp until we can integrate DRB data
-            writer.writerow(["Title", "Timestamp", "Edition_ID"])
-            for download_event in download_events:
-                writer.writerow(
-                    [download_event.title, download_event.timestamp, download_event.edition_id])
+    def aggregate_interaction_events(self, events):
```

Comment:
This is also a giant code block, would appreciate any ideas on how to break this up!

Reply:
Can't think of anything off the top of my head since it's all operations on data from one dataframe